storage/src/main/java/tech/pegasys/teku/storage/server/DepositStorage.java

```diff
@@ -52,27 +60,41 @@ public void stop() {}
   }
 
   private ReplayDepositsResult replayDeposits() {
-    isSyncingFromDatabase = true;
+    if (!eth1DepositsFromStorageEnabled) {
+      startingBlock = Optional.of(BigInteger.valueOf(-1L));
+      return new ReplayDepositsResult(UnsignedLong.ZERO, false);
+    }
+
     final DepositSequencer depositSequencer =
         new DepositSequencer(eth1EventsChannel, database.getMinGenesisTimeBlock());
     try (Stream<DepositsFromBlockEvent> eventStream = database.streamDepositsFromBlocks()) {
       eventStream.forEach(depositSequencer::depositEvent);
     }
     depositSequencer.depositsComplete();
-    isSyncingFromDatabase = false;
-    return depositSequencer.done();
+    ReplayDepositsResult result = depositSequencer.done();
+    if (result.getBlockNumber().isPresent()) {
+      startingBlock =
+          Optional.of(result.getBlockNumber().get().plus(UnsignedLong.ONE).bigIntegerValue());
+    } else {
+      startingBlock = Optional.of(BigInteger.valueOf(-1L));
+    }
+    return result;
+  }
+
+  private boolean shouldProcessEvent(final BigInteger blockNumber) {
+    return startingBlock.isPresent() && startingBlock.get().compareTo(blockNumber) < 0;
```

Comment:
This probably works out but the thread safety is questionable and it's causing IntelliJ to flag warnings.  Better would be to only load the current value of `startingBlock` once.
```suggestion
    return startingBlock.map(block -> block.compareTo(blockNumber) < 0).orElse(false)
```

Reply:
I'd be tempted to make this take an `UnsignedLong` as parameter and do the `.bigIntegerValue()` once in here instead of in the two places it's called.